### ISSUE TITLE
Question for multi, client cache for keys in cluster mode and some recommendations for Medis client

### ISSUE THREAD

**Issue by RuofeiLv** [luin/ioredis#409]
**Question 1**
redis.multi().set('foo'，'value A').set('foo', 'new value').exec(function (err, results) {
});
In the example above( the syntax and arguments are all valid),  if this transaction fails, do we can get the error from function (err, results) {} err object? It's hard to simulate errors because redis seems never fail.

**Quesion 2**
In redis documentation redis cluster tutorials, it says the the redis-cli cluster support is very basic so it always uses the fact that Redis Cluster nodes are able to redirect a client to the right node. A serious client is able to do better than that, and cache the map between hash slots and nodes addresses, to directly use the right connection to the right node. (https://redis.io/topics/cluster-tutorial), So I wanna ask if ioredis implement this function and it seems not.... 

**Recommendations**
PS: I am your Medis user(I bought it from AppStore), two recommendations are to add multiple instance connection at the same time(like roboMongo client for MongoDB) and make multiple selections for keys or values for quick delete.

**Comment by luin** [comment]
**Question 1**
ioredis follows exactly what the Redis server does. When the syntax and arguments are all valid, the transaction will never fail (thus the `err` object will always be `null`). Instead, if some of the commands fail, the error will be returned via `results`. Details: https://redis.io/topics/transactions#errors-inside-a-transaction

**Question 2**
Slot cache is a must-have feature for a redis client to be used in production. So given ioredis is widely used by many companies, we of course implement it. Refer to `getInfoFromNode` function and the handler for `MOVED`, `ASK` error for details.

Thank you for purchasing Medis 😆! Pardon me but I'm wondering what does "add multiple instance connection at the same time" mean.

**Comment by RuofeiLv** [comment]
Thank you for replying.

**Question 1**
I am  gonna check the error handling topic in detail

**Question 2**
I am gonna check the source code about getInfoFromNode, I am curious about how it was implemented, a small question before diving into the source code -- slot cache means the cache for
the map between hash slots and nodes addresses, so basiclly the procedures i guess are

1. Next time we have a new key in client, we firstly calculate the key's slot by using CRC16 of the key modulo 16384
2. Now we get the hash slot and since we have the cache map between hash slots and nodes addresses, then we have the node addresses.
3. Client send the requests to the that node addresses.

4.Redis server get the requests from client, then the redis server has to caculate again about the hash slot of the key to check if this node can receive this request, so this recalcaulation here for hash slot is verbose and useless, right?     However, for server itself, it has to check again for data or system safety. But calculation for key's hash slot do happen twice.


Another question is about the data safety of the redis cluster, I use the redis for two purpose, database and the cache. For database, I store the data such as user follower-following relationship and timeline for each user, I need to ensure the 99.99% safety of the data. So If I follow the recommendation of official configuation, I need to have several slaves for each master, but do we need to turn on or turn of the persistence for master such like AOF? If I turn on the AOF, then I definitely want to restore my data from AOF because it seems to have less possibility for data lose(no more that 1 second if i set the time limit), but for turning off the AOF, the time is not guaranteed(0-N? seconds), so any ideas here?   For cache purpose, I dont' care any data lose. HAHA.



**Recommendation**
Yes, I mean that is the new connection tab or new connection window function that I have found them on File->New connection tab || New connection window !
